### PR TITLE
Update retirement to fetch Segment Ecom ID into config when configured

### PR DESCRIPTION
**Pull request by bmedx**

### CODE REVIEW

File: tubular/scripts/retire_one_learner.py [hunk 1]
```diff
@@ -129,6 +129,22 @@ def _get_learner_and_state_index_or_exit(config, username):
...
+    here than to pass all of the config down to SegmentApi.
+    """
+    try:
+        response = config['ECOMMERCE'].get_tracking_key(learner)
+        return response['ecommerce_tracking_id']
+    except HttpNotFoundError:
+        LOG('Learner {} not found in Ecommerce. Setting Ecommerce Segment ID to None'.format(learner))
+        return None
+    except Exception as exc:  # pylint: disable=broad-except
+        FAIL_EXCEPTION(ERR_SETUP_FAILED, 'Unexpected error fetching Ecommerce tracking id!', text_type(exc))
```
Comment (doctoryes) [inline tubular/scripts/retire_one_learner.py:144]: Will the learner username be included in the exception here?

File: tubular/scripts/retire_one_learner.py [hunk 2]
```diff
@@ -129,6 +129,22 @@ def _get_learner_and_state_index_or_exit(config, username):
         FAIL_EXCEPTION(ERR_SETUP_FAILED, 'Unexpected error fetching user state!', text_type(exc))
 
 
+def _get_ecom_segment_id(config, learner):
```
Comment (doctoryes) [inline tubular/scripts/retire_one_learner.py:132]: Is this method being covered in tests?

File: tubular/tests/test_segment_api.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ def test_suppress_and_delete_success():
     Test simple success case
     """
     projects_to_retire = ['project_1', 'project_2']
-    learner = {'id': 1, 'ecommerce_id': 'ecommerce-20', 'original_username': 'test_user'}
+    learner = {'id': 1, 'ecommerce_segment_id': 'ecommerce-20', 'original_username': 'test_user'}
```
Comment (doctoryes) [inline tubular/tests/test_segment_api.py:33]: So the `ecommerce_id` wasn't being referenced anywhere? Only in `IDENTIFYING_KEYS`?

File: tubular/scripts/retire_one_learner.py (hunk 1 above)
Comment (bmedx) [reply tubular/scripts/retire_one_learner.py:144]: Probably not, but at this point in the Jenkins process the task itself should have the username of the retiring learner.

File: tubular/tests/test_segment_api.py (hunk 3 above)
Comment (bmedx) [reply tubular/tests/test_segment_api.py:33]: Correct, that is it's sole purpose

File: tubular/scripts/retire_one_learner.py (hunk 2 above)
Comment (bmedx) [reply tubular/scripts/retire_one_learner.py:132]: Given that the API call itself will have to be mocked out there's precious little to test, but I can exercise the exception paths. I'll write up some tests for that.

Comment (doctoryes) [reply tubular/scripts/retire_one_learner.py:132]: Feel free to make the call not to do so if you feel there's little value in it.

File: tubular/scripts/retire_one_learner.py [hunk 4]
```diff
@@ -129,6 +129,22 @@ def _get_learner_and_state_index_or_exit(config, username):
...
+    """
+    Calls Ecommerce to get the ecom-specific Segment tracking id that we need to retire.
+    This is only available from Ecommerce, unfortunately, and makes more sense to handle
+    here than to pass all of the config down to SegmentApi.
+    """
+    try:
+        response = config['ECOMMERCE'].get_tracking_key(learner)
+        return response['ecommerce_tracking_id']
```
Comment (pwnage101) [inline tubular/scripts/retire_one_learner.py]: nit: how about just returning the tracking key rather than the entire response dict?

**Review by pwnage101 — APPROVED** [review]
see one nit.

**Pull request merged**